### DISCUSSION TITLE
Support for alerts

### DISCUSSION THREAD

**Discussion by cqueinnec** [grafana/grafana-infinity-datasource#133] [Ideas]
Hi,

Are there any plans to activate alerting support for this datasource? I'm mainly interested in alerts for the GraphQL datasource.

Currently, the message "The datasource does not support alerting queries" appears when trying to tie alerts to Infinity panels.

As far as I can see from docs, the plugin should expose an `alerting` property (both for the new [unified](https://grafana.com/docs/grafana/latest/alerting/unified-alerting/#alert-evaluation) alerting system and for the [legacy](https://grafana.com/docs/grafana/latest/alerting/old-alerting/#alert-evaluation) one). However I don't know if this is as easy as exposing this property, or if specific code should be adjusted in the plugin to make it work. Maybe you've already investigated this!

Thanks for the great plugin :-)

**Comment by yesoreyeram** [comment]
Hey.. thanks for trying the plugin. Adding alerting means the entire codebase should be migrated to backend (go).  Currently the plugin is mostly frontend. Supporting alert require a decent amount of migration. So not anytime soon.

The property `alerting: true` only applicable for the plugins that are having backend query. We are in the process of migrating.

**Reply by spark2ignite** [reply]
Great plug-in! What's the latest on the alerting support?

**Reply by yesoreyeram** [reply]
Hi there... Thanks for the kind words. Upcoming version 1.0 will have some sort of experimental support for this. In a month or two.

**Comment by yesoreyeram** [comment]
1.0.0-beta.1 is now **[available](https://github.com/yesoreyeram/grafana-infinity-datasource/releases/tag/v1.0.0-beta.1)** with alerting support. 

With JSON query type, one need to choose `backend` as a parser to get the alerts working. Example shown below for setting up alert.

<img width="1792" alt="image" src="https://user-images.githubusercontent.com/153843/188924894-4b1f3d3d-2857-484e-916c-d09cb8d4dd5c.png">

Example API used : https://thingspeak.com/channels/38629/feed.json

If anyone interested, feel free to give a try and report any issues.
